src/rules-engine/Property.js

```diff
@@ -40,8 +39,8 @@ class Property extends EventEmitter {
     this.name = parts[parts.length - 1];
 
     this.onMessage = this.onMessage.bind(this);
-    this.ws = null;
-    this.id = Math.floor(Math.random() * 100000);
+    let thingHref = this.href.split('/properties')[0];
```

Comment:
Is there some way we can avoid hard coding this string? We might not be able to assume this URL structure for native web things.

Reply:
Yes, I'll file a follow-up issue since it will also involve migrating any existing rules.